### PR TITLE
CMake: Use GNUInstallDirs for install destination

### LINKED ISSUE #188: The `* .dll` files installed in `lib` directory (should be `bin`)
After installation on DLL-based platform, the `*.dll` part of shared libraries will be placed in `lib` subdirectory, but it should be the `bin` directory.

### PR DESCRIPTION
**Pull request by podsvirov**
Closes #188.

### CODE REVIEW

**Comment by jmr** [comment]
Can you add something to README.md about how to configure install locations?

**Comment by podsvirov** [comment]
> Can you add something to README.md about how to configure install locations?

Done. Please review again (sorry for possible  lingual mistakes).

File: README.md [hunk 1]
```diff
@@ -88,13 +88,44 @@ cd build
...
+make install
+```
+
+Prefix it with `sudo` if needed:
+
+```
+sudo make install
+```
+
+_NOTE_: The `uninstall` target missed but `install_manifest.txt` may be helpfull.
```
Comment (jmr) [inline README.md:111]: How about "there is not `uninstall` target but ..."?

File: README.md [hunk 2]
```diff
@@ -88,13 +88,44 @@ cd build
...
+
+Several suffix variables used for some file groups:
+
+Variable | Description
+-------- | -----------
+`CMAKE_INSTALL_INCLUDEDIR` | For header files
+`CMAKE_INSTALL_BINDIR`     | For executables and `*.dll` files on `DLL`-based platforms
+`CMAKE_INSTALL_LIBDIR`     | For library files (`*.so`, `*.a`, `*.lib` etc)
+
+To edit cached variables type:
```
Comment (jmr) [inline README.md:123]: What are cached variables? I'm not sure this is going to help without further explanation.

**Review by jmr — CHANGES_REQUESTED** [review]

**Pull request merged**